mesonbuild/coredata.py

```diff
@@ -747,48 +682,129 @@ def parse_cmd_line_options(args):
+                                choices=['plain', 'debug', 'debugoptimized', 'release', 'minsize', 'custom']),
+    'strip':      BuiltinOption(UserBooleanOption, 'Strip targets on install', False),
+    'unity':      BuiltinOption(UserComboOption, 'Unity build', 'off', choices=['on', 'off', 'subprojects']),
+    'prefix':     BuiltinOption(UserStringOption, 'Installation prefix', default_prefix()),
+    'libdir':     BuiltinOption(UserStringOption, 'Library directory', default_libdir()),
+    'libexecdir': BuiltinOption(UserStringOption, 'Library executable directory', default_libexecdir()),
+    'bindir':     BuiltinOption(UserStringOption, 'Executable directory', 'bin'),
+    'sbindir':    BuiltinOption(UserStringOption, 'System executable directory', 'sbin'),
+    'includedir': BuiltinOption(UserStringOption, 'Header file directory', 'include'),
+    'datadir':    BuiltinOption(UserStringOption, 'Data file directory', 'share'),
+    'mandir':     BuiltinOption(UserStringOption, 'Manual page directory', 'share/man'),
+    'infodir':    BuiltinOption(UserStringOption, 'Info page directory', 'share/info'),
+    'localedir':  BuiltinOption(UserStringOption, 'Locale data directory', 'share/locale'),
+    'sysconfdir':      BuiltinOption(UserStringOption, 'Sysconf data directory', 'etc'),
+    'localstatedir':   BuiltinOption(UserStringOption, 'Localstate data directory', 'var'),
+    'sharedstatedir':  BuiltinOption(UserStringOption, 'Architecture-independent data directory', 'com'),
+    'werror':          BuiltinOption(UserBooleanOption, 'Treat warnings as errors', False),
+    'warning_level':   BuiltinOption(UserComboOption, 'Compiler warning level to use', '1', choices=['0', '1', '2', '3']),
+    'layout':          BuiltinOption(UserComboOption, 'Build directory layout', 'mirror', choices=['mirror', 'flat']),
+    'default_library': BuiltinOption(UserComboOption, 'Default library type', 'shared', choices=['shared', 'static', 'both']),
+    'backend':         BuiltinOption(UserComboOption, 'Backend to use', 'ninja', choices=backendlist),
+    'stdsplit':        BuiltinOption(UserBooleanOption, 'Split stdout and stderr in test logs', True),
+    'errorlogs':       BuiltinOption(UserBooleanOption, "Whether to print the logs from failing tests", True),
+    'install_umask':   BuiltinOption(UserUmaskOption, 'Default umask to apply on permissions of installed files', '022'),
+    'auto_features':   BuiltinOption(UserFeatureOption, "Override value of all 'auto' features", 'auto'),
+    'optimization':    BuiltinOption(UserComboOption, 'Optimization level', '0', choices=['0', 'g', '1', '2', '3', 's']),
+    'debug':           BuiltinOption(UserBooleanOption, 'Debug', True),
+    'wrap_mode':       BuiltinOption(UserComboOption, 'Wrap mode', 'default', choices=['default', 'nofallback', 'nodownload', 'forcefallback']),
+    'pkg_config_path': BuiltinOption(UserArrayOption, 'List of additional paths for pkg-config to search', [], separate_cross=True),
```

Comment:
This should be added to https://github.com/mesonbuild/meson/blob/master/docs/markdown/Builtin-options.md

Reply:
Yes, we should also add a unit test for it so that it doesn't happen again.

Reply:
@nirbheek I could write such unit test, but that will require the exposure of global `builtin_options` var through some means (according to my limited knowledge it's impossible to access global vars directly from other files/scripts)

Reply:
See: `run_unittests.py:DataTests()`, we already have a test for the compiler options. The same can be done for builtin options too.

Reply:
See #5233